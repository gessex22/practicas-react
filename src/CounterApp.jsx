import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const HandleAdd = () => {
    setCounter(counter + 1);
  };

  const resetCounter = () => {
    setCounter(value);
  };
  const minusCounter = () => {
    setCounter(counter - 1);
  };

  return (
    <>
      <h1> Counter App</h1>
      <h2> {counter}</h2>

      <button onClick={HandleAdd}> addcion +1</button>
      <button onClick={minusCounter}> sustraccion -1</button>
      <button onClick={resetCounter}> reset +1</button>
    </>
  );
};
