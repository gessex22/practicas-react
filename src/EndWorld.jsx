import PropTypes from "prop-types";

export const NdelWorld = ({ title, time }) => {
  return (
    <>
      <h1>{title} </h1>
      <p> Fin del mundo desde un modulo exportado</p>
      <p> {time + 1}</p>
    </>
  );
}

NdelWorld.propType = {
title: PropTypes.string.isRequired,
time: PropTypes.number.isRequired

}


NdelWorld.defaultProps = {

title : 'not title',
time: 99999
}





