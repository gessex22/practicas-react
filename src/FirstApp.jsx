import PropTypes from "prop-types"


export const FirstApp = ({title,subtitle, name})=> {


    return(
    
        <>
            <h1 data-testid="test-title">{ title}</h1>
            <p>{subtitle}</p>
             <p>{subtitle}</p>
            <p>{name}</p>
        
        </>
    
    
    )


}

FirstApp.propTypes = {

    name : PropTypes.string.isRequired
}


