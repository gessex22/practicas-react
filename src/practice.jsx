import PropTypes from "prop-types"


export const Practice = ({sentence})=> {

    const sumaDom = (event) =>{

    console.log( event)
    sentence = 1000
    
    }


    return(
    
        <>
            <h1>Practice </h1>
            <h2> {sentence }</h2>
            <button onClick={ sumaDom} > sumn </button>
        
        </>
    
    
    )


}

Practice.propTypes = {

    sentence : PropTypes.string
}


