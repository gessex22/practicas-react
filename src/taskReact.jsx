import PropTypes from "prop-types";

export const Task = ({ value }) => {
  return (
    <>
      <h1> CounterAPP</h1>
      <h2> {value}</h2>
    </>
  );
};

Task.propTypes = {
  value: PropTypes.number,
};
