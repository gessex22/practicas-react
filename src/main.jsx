import React from 'react';
import ReactDOM from 'react-dom/client';
// import { NdelWorld } from './EndWorld';

import './main.css'
import { CounterApp } from './CounterApp'
//import { Practice } from './practice';
//import { func } from 'prop-types';

 

// function EndWorl() {

//  return (<h1>EL verdader N del world</h1>);

// }


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode> 
     < CounterApp /> 
    </React.StrictMode>
)