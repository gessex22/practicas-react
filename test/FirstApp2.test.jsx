import { render, screen} from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas en firstApp organizado' , ()=>{

    const title = 'Pruebas'
    const subtitle = 'soy un subtitle'



    test('Match con snapshot', ()=>{
    
        const {container } = render( <FirstApp  title={title}/>)
        expect(container).toMatchSnapshot()
    
    })

    test('DEbe mostrar el msg quien soy', ()=>{
    
      render( <FirstApp  title={title}/>)
      expect(screen.getByText(title)).toBeTruthy()
     
    })

    test('Debe de mostrsar el title en un H1' , ()=>{
    
        render( <FirstApp  title={title}/>)
        expect(screen.getByRole('heading', {level:1}).innerHTML ).toContain(title)
    
    })

    test('Debe mostrarse el subtitle enviado por props', ()=>{
    
        render(<FirstApp 
            title={ title}
            subtitle={ subtitle }
        
        />) 
        expect( screen.getAllByText(subtitle).length ).toBe(2)
    })



})