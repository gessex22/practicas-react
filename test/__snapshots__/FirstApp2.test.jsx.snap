// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en firstApp organizado Match con snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Pruebas
  </h1>
  <p />
  <p />
  <p />
</div>
`;
